folding-schemes/src/folding/nova/mod.rs

```diff
@@ -166,17 +166,21 @@ impl<C: CurveGroup> Witness<C>
 where
     <C as Group>::ScalarField: Absorb,
 {
-    pub fn new(w: Vec<C::ScalarField>, e_len: usize) -> Self {
-        // note: at the current version, we don't use the blinding factors and we set them to 0
-        // always.
+    pub fn new(
+        w: Vec<C::ScalarField>,
+        rW: C::ScalarField,
+        e_len: usize,
+        rE: C::ScalarField,
+    ) -> Self {
         Self {
             E: vec![C::ScalarField::zero(); e_len],
-            rE: C::ScalarField::zero(),
+            rE,
             W: w,
-            rW: C::ScalarField::zero(),
+            rW,
```

Comment:
if I understand correctly, this change has been introduced in the last update of the PR. Related to https://github.com/privacy-scaling-explorations/sonobe/pull/129/files#r1690865824 , could this be done as in hypernova? (and not reusing the blindings across different instances)

Reply:
Just commented about this above, tell me if that's what you had in mind!

Reply:
This is a very minor thing, feel free to skip it:

as in https://github.com/privacy-scaling-explorations/sonobe/pull/129#discussion_r1682099466 and https://github.com/privacy-scaling-explorations/sonobe/pull/129#discussion_r1690865824 , I think this could be done as in HyperNova implementation for the blinding params, for example:
moving the current lines 694-702 into `Witness::new`
```rust
        // if hiding, sample new blinding values
        let (rW, rE) = if H {
            (
                C1::ScalarField::rand(&mut rng),
                C1::ScalarField::rand(&mut rng),
            )
        } else {
            (C1::ScalarField::zero(), C1::ScalarField::zero())
        };
```
so that in this way there is no need to pass the random values into the `Witness::new`, and is part of the 'new witness creation' logic to generate the blinding factors if hiding is activated.


The reasoning is having similar approach in both nova&hypernova, but also because there is no need to choose the random blindings outside the `Witness` creation, so that the `Witness::new` interface can be simplier and with less logic done before (the random values creation with the `if H`).

Reply:
Yes I agree, I did two commits that implement it this way. The first one adds a `const H: bool` generic to the `new` method when instantiating a new witness. The second one adds a new method for generating dummy witnesses. This is useful to avoid passing an `rng` object when instantiating dummy instances.